examples/minipc/ChassisTest.cc

```diff
@@ -0,0 +1,187 @@
+  the chassis. 
+
+  On the Jetson side, the following scripts in robot_driver work with this
+  example: chassis_test.py, cmd_vel_adjuster.py, and keyboard_teleop.py.
+
+  (Note that if you are using cmd_vel_adjuster.py and keyboard_teleop.py, you
+  will need to run robot.py as well. See each script for more information.)
+*/
+
+#define RX_SIGNAL (1 << 0)
+#define DATA_READY_SIGNAL (1 << 1)
+extern osThreadId_t defaultTaskHandle;
+
+static display::RGB* led = nullptr;
+bsp::CAN* can = nullptr;
+control::MotorCANBase* fl_motor = nullptr;
+control::MotorCANBase* fr_motor = nullptr;
+control::MotorCANBase* bl_motor = nullptr;
+control::MotorCANBase* br_motor = nullptr;
+
+control::Chassis* chassis = nullptr;
+remote::DBUS* dbus = nullptr;
+
+osEventFlagsId_t chassis_flag_id;
+
+static float vx = 0;
+static float vy = 0;
+static float vw = 0;
+
+class CustomUART : public bsp::UART {
+ public:
+  using bsp::UART::UART;
+
+ protected:
+  /* notify application when rx data is pending read */
+  void RxCompleteCallback() override final { osThreadFlagsSet(defaultTaskHandle, RX_SIGNAL); }
+};
+
+const osThreadAttr_t chassisTaskAttribute = {.name = "chassisTask",
+                                             .attr_bits = osThreadDetached,
+                                             .cb_mem = nullptr,
+                                             .cb_size = 0,
+                                             .stack_mem = nullptr,
+                                             .stack_size = 256 * 4,
+                                             .priority = (osPriority_t)osPriorityNormal,
+                                             .tz_module = 0,
+                                             .reserved = 0};
+osThreadId_t chassisTaskHandle;
+
+void chassisTask(void* argument) {
+  UNUSED(argument);
+  uint32_t flags;
+  control::MotorCANBase* motors[] = {fl_motor, fr_motor, bl_motor, br_motor};
+
+  float vx_keyboard = 0.0;
+  float vy_keyboard = 0.0;
+  float wz_keyboard = 0.0;
+
+  while (true) {
+    // Keyboard control / Navigation mode
+    if (dbus->swr == remote::UP) {
+      flags = 0;
+      flags = osEventFlagsWait(chassis_flag_id, DATA_READY_SIGNAL, osFlagsWaitAny, 50);
+
+      // When timeout it returns -2 so we need extra checks here
+      if (flags != osFlagsErrorTimeout && flags & DATA_READY_SIGNAL) {
+        vx_keyboard = clip<float>(vx, -1200, 1200);
+        vy_keyboard = clip<float>(vy, -1200, 1200);
+        wz_keyboard = vw;
+
+        chassis->SetSpeed(vx_keyboard, vy_keyboard, wz_keyboard);
+        chassis->Update(false, 30, 20, 60);
+
+        led->Display(display::color_green);
+      } else {
+        // if timeout (no packet, stop the motor)
+        fl_motor->SetOutput(0);
+        fr_motor->SetOutput(0);
+        bl_motor->SetOutput(0);
+        br_motor->SetOutput(0);
+
+        led->Display(display::color_red);
+      }
+      control::MotorCANBase::TransmitOutput(motors, 4);
+      osDelay(10);
```

Comment:
Also, this and line 122/123 can be joined together.

Reply:
As a side effect helps to make `led->Display` more consistent :)

Reply:
Fixed.